doc/api/module.md

```diff
@@ -217,6 +217,39 @@ added: v22.8.0
 * Returns: {string|undefined} Path to the [module compile cache][] directory if it is enabled,
   or `undefined` otherwise.
 
+### `module.getPackageJSON(startPath[, everything])`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+> Stability: 1.1 - Active Development
+
+* `startPath` {string} Where to start looking
+* `everything` {boolean} Whether to return the full contents of the found package.json
+* Returns: {undefined | {
+  data: {
+  name?: string,
+  type?: 'commonjs' | 'module' | 'none',
+  exports?: string | string\[] | Record\<string, unknown>,
+  imports?: string | string\[] | Record\<string, unknown>,
+  \[key: string]?: unknown,
+  },
+  path: URL\['pathname'],
+  }}
+
+In addition to being available to users, this utility is used internally when
```

Comment:
I don't think it's a good idea to talk about what is used internally in the docs - users would assume that this can be patchable, like what you do with many other module APIs. It also hints that internally, the consumer of this function is JavaScript, which isn't necessarily the case if we port the loader to C++ again for performance.

Reply:
Good call :)